File: Servidor/src/main/java/com/example/Package/models/Guide.java
package com.example.Package.models;

import java.time.LocalDate;

public class Guide {
    private int id;
    private String nombre;
    private double calificacion;
    private int edad;
    private LocalDate fechaNacimiento;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getCalificacion() {
        return calificacion;
    }

    public void setCalificacion(double calificacion) {
        this.calificacion = calificacion;
    }

    public int getEdad() {
        return edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public LocalDate getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(LocalDate fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public Guide(int id, String nombre, double calificacion, int edad, LocalDate fechaNacimiento) {
        this.id = id;
        this.nombre = nombre;
        this.calificacion = calificacion;
        this.edad = edad;
        this.fechaNacimiento = fechaNacimiento;
    }

    @Override
    public String toString() {
        return "Guide{" +
                "id=" + id +
                ", nombre='" + nombre + '\'' +
                ", calificacion=" + calificacion +
                ", edad=" + edad +
                ", fechaNacimiento=" + fechaNacimiento +
                '}';
    }
}

File: client-react/src/modelo/Guide.ts
export class Guide {
    private id: number;
    private nombre: string;
    private calificacion: number;
    private edad: number;
    private fechaNacimiento: string;
  
    // Constructor para inicializar las propiedades
    constructor(
      id: number,
      nombre: string,
      calificacion: number,
      edad: number,
      fechaNacimiento: string
    ) {
      this.id = id;
      this.nombre = nombre;
      this.calificacion = calificacion;
      this.edad = edad;
      this.fechaNacimiento = fechaNacimiento;
    }
  
    // Getters y Setters
    get getid(): number {
      return this.id;
    }
  
    set setid(value: number) {
      this.id = value;
    }
  
    get getnombre(): string {
      return this.nombre;
    }
  
    set setnombre(value: string) {
      this.nombre = value;
    }
  
    get getcalificacion(): number {
      return this.calificacion;
    }
  
    set setcalificacion(value: number) {
      this.calificacion = value;
    }
  
    get getedad(): number {
      return this.edad;
    }
  
    set setedad(value: number) {
      this.edad = value;
    }
  
    get getfechaNacimiento(): string {
      return this.fechaNacimiento;
    }
  
    set setfechaNacimiento(value: string) {
      this.fechaNacimiento = value;
    }
  
    // Método para imprimir la información del guía
    toStringg(): string {
      return `ID: ${this.id}, Nombre: ${this.nombre}, Calificación: ${this.calificacion}, Edad: ${this.edad}, Fecha de Nacimiento: ${this.fechaNacimiento}`;
    }
  }
  